changelog/front-commerce-3.4-2.29.mdx

```diff
@@ -0,0 +1,471 @@
+upgrade to new versions without breaking their existing codebase.**
+
+These APIs also ensure that people can contribute to the product's growth by
+developing plugins, extensions, and integrations.
+
+We wanted to wait for the Vite migration to stabilize as it impacted many parts
+of our codebase. Now that it's done, we are happy to introduce new stable APIs
+in the core of Front-Commerce.
+
+### GraphQL Module stable API
+
+We reworked how GraphQL modules are registered to introduce a clearer and more
+declarative `graphql.modules` registration API for extensions and app authors.
+
+While conceptually similar to the way a GraphQL module was registered in 2.x,
+the new API more clearly separates the concerns of a GraphQL module:
+
+- the **module definition** (using `createGraphQLModule`) is loaded with your
+  `front-commerce.config.ts` configuration declaration. It is a plain Javacript
+  object that declares the responsibilities of the module and its role in the
+  application. It is where you'll define the schema, name, and dependencies of
+  the module.
+- the **module runtime** (using `createGraphQLRuntime`) is a lazy-loaded module
+  that is only loaded when the application starts. It is where you'll define the
+  `resolvers` and `contextEnhancer` of your module, that is, the code that will
+  be executed when a GraphQL query is made.
+
+We hope this new API will make it easier for you to understand and maintain your
+GraphQL modules. We implemented several transformers in the
+`@front-commerce/codemod` 3.4 migration, to help you migrate your codebase to
+this new API.
+
+Read the
+[migration guide](/docs/3.x/upgrade/migration-guides/3.3-3.4#from-graphqlunstable_module-to-graphqlmodule)
+for more information about these changes.
+
+### Sitemap API
+
+Front-Commerce 3.4 introduces a revamped way to generate sitemaps. The new API
+is more flexible and is based on Remix routes instead of CRON jobs.
+
+Here are some highlights of the new API:
+
+- Front-Commerce will use your defined routes to generate the sitemap
+- You can customize the content of the sitemap at route level using your route
+  `handle` export
+- We provide a `createSeoHandle()` helper to achieve the most common use cases
+  with a few lines of code
+- Dynamic routes can leverage **Sitemap fetchers** from registered extensions to
+  generate URLs based on data from remote systems (like Magento or Contentful)
+
+Read the [Customize the sitemap](/docs/3.x/guides/customize-the-sitemap) guide
+to learn more about the new sitemap API.
+
+Our core modules `@front-commerce/theme-chocolatine`, `@front-commerce/magento1`
+and `@front-commerce/magento2` have has been updated to use this new API.
+
+:::note Previous work
+
+This API was inspired by the work of
+[@balavishnuvj/remix-seo](https://www.npmjs.com/package/@balavishnuvj/remix-seo)
+and the [NASA GCN fork](https://github.com/nasa-gcn/gcn.nasa.gov). Thanks to
+them for their contributions!
+
+:::
```

Comment:
@real34 if we get https://github.com/front-commerce/developers.front-commerce.com/pull/889 reviewed and merged, then you could possibly use the artwork there to add a brief explanation of the Sitemap Service api?

https://deploy-preview-889--heuristic-almeida-1a1f35.netlify.app/docs/3.x/guides/customize-the-sitemap#creating-and-registering-a-custom-fetchers

Reply:
I duplicated it for the changelog entry. This way, the doc could evolve in the future and both are independent.